discovery/discovery.go

```diff
@@ -59,68 +60,68 @@ func ProvidersFromConfig(cfg config.ServiceDiscoveryConfig, logger log.Logger) m
 	}
 
 	for i, c := range cfg.DNSSDConfigs {
-		app("dns", i, dns.NewDiscovery(c, logger))
+		app("dns", i, dns.NewDiscovery(c, log.With(logger, "discovery", "dns")))
 	}
 	for i, c := range cfg.FileSDConfigs {
-		app("file", i, file.NewDiscovery(c, logger))
+		app("file", i, file.NewDiscovery(c, log.With(logger, "discovery", "file")))
 	}
 	for i, c := range cfg.ConsulSDConfigs {
-		k, err := consul.NewDiscovery(c, logger)
+		k, err := consul.NewDiscovery(c, log.With(logger, "discovery", "consul"))
 		if err != nil {
-			logger.Errorf("Cannot create Consul discovery: %s", err)
+			level.Error(logger).Log("msg", "Cannot create Consul discovery", "err", err)
 			continue
 		}
 		app("consul", i, k)
 	}
 	for i, c := range cfg.MarathonSDConfigs {
-		m, err := marathon.NewDiscovery(c, logger)
+		m, err := marathon.NewDiscovery(c, log.With(logger, "discovery", "marathon"))
 		if err != nil {
-			logger.Errorf("Cannot create Marathon discovery: %s", err)
+			level.Error(logger).Log("msg", "Cannot create Marathon discovery", "err", err)
 			continue
 		}
 		app("marathon", i, m)
 	}
 	for i, c := range cfg.KubernetesSDConfigs {
-		k, err := kubernetes.New(logger, c)
+		k, err := kubernetes.New(log.With(logger, "discovery", "k8s"), c)
 		if err != nil {
-			logger.Errorf("Cannot create Kubernetes discovery: %s", err)
+			level.Error(logger).Log("msg", "Cannot create Kubernetes discovery", "err", err)
 			continue
 		}
 		app("kubernetes", i, k)
 	}
 	for i, c := range cfg.ServersetSDConfigs {
-		app("serverset", i, zookeeper.NewServersetDiscovery(c, logger))
+		app("serverset", i, zookeeper.NewServersetDiscovery(c, log.With(logger, "discovery", "zookeeper")))
```

Comment:
Yea, I'm kind torn on this. Isn't severset just the format? OTOH, the discovery has a hard dependency on zookeeper. So `zookeeper_serverset` might be the accurate description?